pyro/distributions/torch/one_hot_categorical.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import absolute_import, division, print_function
+
+import torch
+from torch.autograd import Variable
+
+from pyro.distributions.one_hot_categorical import OneHotCategorical as _OneHotCategorical
+from pyro.distributions.torch_wrapper import TorchDistribution
+from pyro.distributions.util import copy_docs_from, get_clamped_probs
+
+
+@copy_docs_from(_OneHotCategorical)
+class OneHotCategorical(TorchDistribution):
+    enumerable = True
+
+    def __init__(self, ps=None, logits=None, *args, **kwargs):
+        ps = get_clamped_probs(ps, logits, is_multidimensional=True)
+        torch_dist = torch.distributions.Categorical(ps)  # TODO switch to OneHotCategorical
+        x_shape = ps.shape
+        event_dim = 1
+        super(OneHotCategorical, self).__init__(torch_dist, x_shape, event_dim, *args, **kwargs)
+
+    def sample(self):
+        ps = self.torch_dist.probs.data
+        zero = ps.new(self._sample_shape + ps.shape).zero_()
+        indices = super(OneHotCategorical, self).sample().data
+        if indices.dim() < zero.dim():
```

Comment:
Is this for when we have scalar support in PyTorch? `sample` currently should return a non-zero dim tensor/variable.

Reply:
I believe this is to compensate for Pyro distributions adding an extra dim in all cases (for consistency), whereas I believe PyTorch currently adds an extra dim only to tensors that would be scalars but can't yet be. But honestly I just type random stuff until tests pass 😊